commons/src/main/java/com/powsybl/commons/reporter/LoggerReporter.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.reporter;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class LoggerReporter implements Reporter, ReportSeeker {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggerReporter.class);
+
+    private static final String DEFAULT_ROOT_TASK_KEY = "rootTaskKey";
+
+    private final String taskKey;
+    private final String defaultName;
+    private final LoggerReporter parentReporter;
+    private final List<ReportSeeker> childReporters = new ArrayList<>();
+    private final Map<String, Object> taskValues;
+    private final Map<String, Report> reports = new LinkedHashMap<>();
+
+    public LoggerReporter() {
+        this(DEFAULT_ROOT_TASK_KEY, "Root task", Collections.emptyMap(), null);
+    }
+
+    public LoggerReporter(String rootTaskKey, String rootDefaultName, Map<String, Object> taskValues, LoggerReporter parent) {
+        this.taskKey = rootTaskKey;
+        this.defaultName = rootDefaultName;
+        this.taskValues = taskValues;
+        this.parentReporter = parent;
+    }
+
+    @Override
+    public LoggerReporter createChild(String taskKey, String defaultName, Map<String, Object> values) {
+        LoggerReporter childReporter = new LoggerReporter(taskKey, defaultName, values, this);
+        childReporters.add(childReporter);
+        return childReporter;
+    }
+
+    public void export(File file) {
+        try (PrintWriter writer = new PrintWriter(file)) {
+            printTaskReport(this, writer, "");
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```

Comment:
The interfaces have changed, so this is outdated, but in case of IOException in the new equivalent method I throw an UncheckedIOException (in fact I took that behaviour from powsybl-single-line-diagram).